Add additional device info (model, serial number, sw version)

flacjacket opened the pull request:
Read and add additional device information to the hub using the parameters read from the hub.

With this change, migrate the device to use the serial number rather than the name of the device as the unique id.  To facilitate this, split the current `SolaredgeModbusHub` into the `Hub` functionality, which handles reading from the device, and the `SolaredgeModbusCoordinator`, which handles the Home Assistant updates by reading from the `Hub`. This allows the hub to be instatiated in the migration step, which is needed to move the entities over to have the new unique id.

For some reason, doing this migration leaves the old device with no entities in it, so add an empty `async_remove_config_entry_device` function, which makes a "Delete" button show up on the device, allowing the orphan device to be deleted from the UI.

flacjacket commented:
Note that this branch currently includes the commit in #312, since I wanted to use the async api to write this.

binsentsu commented:
Can you fix conflicts? Can you eloborate more on what the # omit NULL terminators changes/fixes in payload.py? Thx

flacjacket commented:
I rebased against current master.

The `payload` change is doing similar logic to the `decode_string` helper function logic that is removed here: https://github.com/binsentsu/home-assistant-solaredge-modbus/pull/314/files#diff-ef1584470671f19eebfa9b5e3d2cf2fca2133bde65d6f8211778a36ab5a4c176L874
Basically the register data for strings has the string itself and then is padded with null bytes, so we need to strip them off. I moved that logic from the helper function into the `decode_string` function on the payload decoder so it could be reused between the battery reading and device information reading.
